airbyte-config/config-models/src/main/java/io/airbyte/config/Configs.java

```diff
@@ -541,13 +559,77 @@ public interface Configs {
    */
   boolean shouldRunConnectionManagerWorkflows();
 
+  /**
+   * Define if the worker is operating as a Control Plane worker, a Data Plane worker, or as a
+   * combined worker. - Control plane workers process tasks related to control-flow, like scheduling
+   * and routing. - Data plane workers process tasks related to data syncing, and are typically
+   * deployed outside of Airbyte's internal network. - Combined workers process all tasks, and are the
+   * default for OSS deployments. - Internal-use only.
+   */
+  WorkerPlane getWorkerPlane();
+
+  // Worker - Control Plane configs
+
+  /**
+   * Return true if the WorkerPlane is CONTROL_PLANE or COMBINED. Internal-use only.
+   */
+  boolean isControlPlaneWorker();
+
+  /**
+   * TEMPORARY: Define a set of connection IDs that should run in Airbyte's AWS Data Plane. - This
+   * should only be set on Control-plane workers, since those workers decide which Data Plane task
+   * queue to use based on connectionId. - Will be removed in favor of the Routing Service in the
+   * future. Internal-use only.
+   */
+  Set<String> connectionIdsForAwsDataPlane();
```

Comment:
nit: can we call this `getconnectionIdsForDataPlane` to minimise leaking Cloud bits?